packages/runtime-dom/src/helpers/useCssVars.ts

```diff
@@ -57,7 +57,16 @@ function setVarsOnVNode(
   if (vnode.shapeFlag & ShapeFlags.ELEMENT && vnode.el) {
     const style = vnode.el.style
     for (const key in vars) {
-      style.setProperty(`--${prefix}${key}`, unref(vars[key]))
+      const valueType = typeof vars[key]
+      if (valueType === 'string' || valueType === 'number') {
```

Comment:
the check should be dev only with `__DEV__` `if (__DEV__ && typeof ...) { warn(...) }`

Reply:
Ok, I think I see my disconnect. I was thinking of trying to prevent the CSS variable fallback problem while you were suggesting allowing it through but providing a warning about the potential problem. Hopefully I have this sorted out now.

I also found that the value being null doesn't cause the CSS variable fallback problem so I made that one not cause the warning as well. After seeing that null causes the CSS variable creation to be skipped (which makes sense) then I can see use cases for actually wanting to make the value null.

Another fun note, passing an array with no entries allows the CSS variable fallback to work, one entry works with a valid CSS property value, and more than one breaks it altogether. I'm not sure why one would use an array in this case though.